x/axelarnet/message_handler_test.go

```diff
@@ -288,6 +295,85 @@ func TestHandleMessage(t *testing.T) {
 			assert.Equal(t, genMsg.Status, nexus.Approved)
 		}).
 		Run(t)
+
+	setFee := func(amount sdk.Int, recipient sdk.AccAddress) {
+		fee := axelarnet.Fee{
+			Amount:    amount.String(),
+			Recipient: recipient.String(),
+		}
+		message.Fee = &fee
+		ics20Packet.Memo = string(funcs.Must(json.Marshal(message)))
+		packet = axelartestutils.RandomPacket(ics20Packet, ibctransfertypes.PortID, sourceChannel, ibctransfertypes.PortID, receiverChannel)
+	}
+
+	whenMessageIsValid.
+		When("fee is negative", func() {
+			setFee(sdk.NewInt(-1000), rand.AccAddr())
+		}).
+		Then("should return ack error", ackError()).
+		Run(t)
+
+	whenMessageIsValid.
+		When("fee is zero", func() {
+			setFee(sdk.ZeroInt(), rand.AccAddr())
+		}).
+		Then("should return ack error", ackError()).
+		Run(t)
+
+	whenMessageIsValid.
+		When("fee is greater than transfer amount", func() {
+			feeAmount := funcs.MustOk(sdk.NewIntFromString(ics20Packet.Amount)).Add(sdk.OneInt())
+			setFee(feeAmount, rand.AccAddr())
+		}).
+		Then("should return ack error", ackError()).
+		Run(t)
+
+	// Fee related tests
+	isAssetRegistered := func(isRegistered bool) func() {
+		return func() {
+			n.IsAssetRegisteredFunc = func(ctx sdk.Context, chain nexus.Chain, denom string) bool {
+				return isRegistered
+			}
+		}
+	}
+
+	isAddressBloked := func(isBlocked bool) func() {
+		return func() {
+			b.BlockedAddrFunc = func(addr sdk.AccAddress) bool {
+				return isBlocked
+			}
+		}
+	}
+
+	whenMessageIsValid.
+		When("fee denom is not registered", isAssetRegistered(false)).
+		When("message with fee", func() {
+			setFee(funcs.MustOk(sdk.NewIntFromString(ics20Packet.Amount)), rand.AccAddr())
+		}).
+		Then("should return ack error", ackError()).
+		Run(t)
+
+	whenMessageIsValid.
+		When("fee denom is registered", isAssetRegistered(true)).
+		When("fee recipient is blocked", isAddressBloked(true)).
+		When("message with fee", func() {
+			setFee(funcs.MustOk(sdk.NewIntFromString(ics20Packet.Amount)), rand.AccAddr())
+		}).
+		Then("should return ack error", ackError()).
+		Run(t)
+
+	whenMessageIsValid.
+		When("fee denom is registered", isAssetRegistered(true)).
+		When("fee recipient is not blocked", isAddressBloked(false)).
+		When("message with fee", func() {
+			setFee(funcs.MustOk(sdk.NewIntFromString(ics20Packet.Amount)), rand.AccAddr())
+		}).
+		Then("should return ack success", func(t *testing.T) {
+			assert.True(t, axelarnet.OnRecvMessage(ctx, k, ibcK, n, b, r, packet).Success())
+			assert.Equal(t, genMsg.Status, nexus.Approved)
+		}).
+		Run(t)
```

Comment:
Test some other cases like, fee amount > transfer amount, and specifically edge case overflows. transfer amount is `1` but fee is max int 256, or fee is more than max int, or fee is negative.

Reply:
it's more so to ensure the implementation doesn't break due to unintentional changes